covid19-dashboard/dc-covid19/src/MetaDataGenerator.ts

```diff
@@ -1,123 +1,178 @@
+    } else {
+        return `${numberWithCommas(value)}`
+    }
 }
 
-
 /**
- * Generates information for the graph to display. For example, the Bar Graph must know things such as 'name',
+ * Generates an object for the information to display.
+ * For example, the Bar Graph must know things such as 'value', 'name',
  * 'onHoverInfo', and the 'text' to display on top of the bar.
  * @param dateToGeoIdToValue: date->geoId->value
  * @param geoIdToPopulation: geoId->Population
  * @param geoIdToName: geoId->name
- * @param absoluteIncrease: absolute increase from one day to the other (optional)
+ * @param dateToGeoIdToAbsolute: absolute number from date X to date Y
  * @param label: Are we talking about cases or deaths?
- * @param contentType: What is the graph showing? Per-Capita? Increase? Absolute?
+ * @param calculationType: calculations performed on data: eg. ['difference']
  */
-export default function generateGraphMetadata(dateToGeoIdToValue: DateToGeoIdToValue, geoIdToPopulation: {geoId: number},
-                                              geoIdToName: {geoId: string}, absoluteIncrease: DateToGeoIdToValue,
-                                              label: string, contentType: string): {date: {geoId: Metadata}} | {} {
-    let output: {date: {geoId: Metadata}} | {} = {}
+function generateMetadata(dateToGeoIdToValue: DateToGeoIdToValue,
+                          geoIdToPopulation: {[geoId: string]: number},
+                          geoIdToName: {[geoId: string]: string},
+                          dateToGeoIdToAbsolute: DateToGeoIdToValue,
+                          label: string,
+                          calculationType: string[]): DateToGeoIdToValue {
+
+    let output: DateToGeoIdToValue | {} = {}
 
     // For every date and geoId in the dataset
     for (let date in dateToGeoIdToValue){
         for (let geoId in dateToGeoIdToValue[date]) {
-            const iterativePopulation: number = geoIdToPopulation[geoId] || 0
+            const iterativePopulation = geoIdToPopulation[geoId]
             // Get the name of the state, and get rid of county or Parish
-            let iterativeRegionName: string = geoIdToName[geoId]?.[0].replace(" County", "").replace(" Parish", "")
-            // iterativeBelongingRegion holds the region that it belongs to, for example geoId/12345 belongs to geoId/12
-            const iterativeBelongingRegion: string = geoIdToName[geoId]?.[1]
-            const iterativeValue: number = dateToGeoIdToValue[date]?.[geoId] || 0
-            const iterativeAbsoluteIncreaseValue: number = absoluteIncrease[date]?.[geoId] || 0
-
-            // If the region not a State, append the State to the name. Example: "Miami, Florida"
-            if (iterativeBelongingRegion.includes("geoId"))
-                iterativeRegionName += ", " + geoIdToName[iterativeBelongingRegion]?.[0]
+            let regionName: string = geoIdToName[geoId]?.[0] || geoId
+
+            regionName.replace(" County", "")
```

Comment:
make "County", "Parish", "Borough" constant, and you will for sure expand the content in the future.

And use a for loop (or fancier way) to replace